lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,513 @@
+import assert, { throws } from 'assert';
+
+import * as connext from '@connext/client';
+import { ConnextStore, FileStorage } from '@connext/store';
+import { IConnextClient } from '@connext/types';
+
+import { SwapClientType, SwapRole, SwapState } from '../constants/enums';
+import { CurrencyInstance } from '../db/types';
+import Logger from '../Logger';
+import swapErrors from '../swaps/errors';
+import SwapClient, {
+  ChannelBalance,
+  ClientStatus,
+  PaymentState,
+  WalletBalance,
+  TradingLimits,
+} from '../swaps/SwapClient';
+import { SwapDeal } from '../swaps/types';
+import { UnitConverter } from '../utils/UnitConverter';
+import { ConnextWallet } from './ConnextWallet';
+import errors, { errorCodes } from './errors';
+import {
+  Channel,
+  ConnextChannelCount,
+  ConnextClientConfig,
+  ConnextInfo,
+  ConnextVersion,
+  TokenPaymentRequest,
+  TokenPaymentResponse,
+} from './types';
+
+const MAX_AMOUNT = Number.MAX_SAFE_INTEGER;
+
+/**
+ * A class representing a client to interact with connext.
+ */
+class ConnextClient extends SwapClient {
```

Comment:
I think this is still missing the code where we listen for incoming payments with unknown preimages to request them from xud?